Update Scala versions to 2.12.11

ijuma opened the pull request.

ijuma committed:
Update Scala versions to 2.12.11

typesafe-tools commented:
To validate this pull request against other sbt modules, please visit <A HREF="https://jenkins.scala-sbt.org:8592/job/sbt-validator/parambuild/?sbt=develop&zinc=pull/751/head&io=develop&librarymanagement=develop&util=develop&website=develop&deploy_to_bintray=false&run_sbt_community=false&verbose_dbuild=false&community_verbose=false">this link.</A>

dwijnand committed:
Upgrade sbt-whitesource to 0.1.18

Without this Travis CI was failing:
https://travis-ci.org/github/sbt/zinc/jobs/671727552

    [error] Caused by: java.lang.RuntimeException: Multiple elements for the same key (org.scala-lang.modules,scala-xml_2.12):
    [error] 	ModuleInfo(org.scala-lang.modules,scala-xml_2.12,1.2.0,Some((Artifact(scala-xml_2.12, bundle, jar, None, Vector(), Some(https://repo1.maven.org/maven2/org/scala-lang/modules/scala-xml_2.12/1.2.0/scala-xml_2.12-1.2.0.jar), Map(), None, false),/home/travis/.cache/coursier/v1/https/repo1.maven.org/maven2/org/scala-lang/modules/scala-xml_2.12/1.2.0/scala-xml_2.12-1.2.0.jar)))
    [error] 	ModuleInfo(org.scala-lang.modules,scala-xml_2.12,1.2.0,Some((Artifact(scala-xml_2.12, bundle, jar, None, Vector(), Some(https://repo1.maven.org/maven2/org/scala-lang/modules/scala-xml_2.12/1.2.0/scala-xml_2.12-1.2.0.jar), Map(), None, false),/home/travis/.cache/coursier/v1/https/repo1.maven.org/maven2/org/scala-lang/modules/scala-xml_2.12/1.2.0/scala-xml_2.12-1.2.0.jar)))
    [error] 	at scala.sys.package$.error(package.scala:30)
    [error] 	at sbtwhitesource.package$KeyByAndMergeSyntax$.$anonfun$keyByAndMerge$2(package.scala:26)

This brings in the following 2 fixes:

* https://github.com/lightbend/sbt-whitesource/pull/68
* https://github.com/lightbend/sbt-whitesource/pull/70

dwijnand committed:
Correctly set compilerBridgeTest's scalaVersion

compilerBridgeTest passes the scala version through a system property:

    Test / javaOptions += {
      s"-Dzinc.build.compilerbridge.scalaVersion=${scalaVersion.value}"
    },

But this was falling back to the default.  Fixing that:

    > inspect compilerBridgeTest/scalaVersion
    [info] Setting: java.lang.String = 2.12.10
    [info] Description:
    [info] 	The version of Scala used for building.
    [info] Provided by:
    [info] 	Global / scalaVersion
    [info] Defined at:
    [info] 	(sbt.Defaults.compileBaseGlobal) Defaults.scala:505
    ...
    > reload
    ...
    > inspect compilerBridgeTest/scalaVersion
    [info] Setting: java.lang.String = 2.12.11
    [info] Description:
    [info] 	The version of Scala used for building.
    [info] Provided by:
    [info] 	ProjectRef(uri("file:/d/zinc/"), "compilerBridgeTest") / scalaVersion
    [info] Defined at:
    [info] 	/d/zinc/build.sbt:558
    ...

dwijnand committed:
Add Coursier path for macOS

dwijnand committed:
Add Coursier path for Win too

... for AppVeyor.

ijuma commented on project/Dependencies.scala:
```diff
@@ -6,7 +6,7 @@ object Dependencies {
   val scala211 = "2.11.12"
-  val scala212 = "2.12.10"
+  val scala212 = "2.12.11"
   val scala213 = "2.13.1"
```
Thanks for the various fixes. Scala 2.13.2 has been released this week, do we want to update that too?

dwijnand replied:
Yeah, good thinking.

dwijnand committed:
Update Scala 2.13 version to 2.13.2

eed3si9n reviewed (approved):
Thanks @ijuma and @dwijnand!
